File: src/connectionPreview.js
import GObject from 'gi://GObject';
import Gtk from 'gi://Gtk';
import Tlg from 'gi://Tlg';

export class ConnectionPreview extends Gtk.Button {
    static {
        GObject.registerClass({
            GTypeName: "TrilogyConnectionPreview",
            Properties: {
                'connection': GObject.ParamSpec.object(
                    'connection',
                    'Connection',
                    'The connection to preview',
                    GObject.ParamFlags.READWRITE | GObject.ParamFlags.CONSTRUCT,
                    Tlg.Connection
                )
            },
            Signals: {
                'selected': {
                    param_types: [Tlg.Connection]
                }
            },
            Template: "resource:///com/rutins/Trilogy/connectionPreview.ui"
        }, this)
    }

    get connection() {
        return this._connection ?? null;
    }

    set connection(conn) {
        if(this._connection == conn) return;
        this._connection = conn;
        this.notify('connection');
    }

    constructor(conn) {
        super({ 'connection': conn });
    }

    on_clicked() {
        this.emit('selected', this.connection);
    }
}

File: src/connection_preview.py
from gi.repository import Gtk, GObject, Tlg

@Gtk.Template(resource_path = "/com/rutins/Trilogy/connectionPreview.ui")
class ConnectionPreview(Gtk.Box):
    __gtype_name__ = 'TrilogyConnectionPreview'
    __gsignals__ = {
        'selected': (GObject.SIGNAL_RUN_LAST, GObject.TYPE_NONE, (Tlg.Connection,))
    }

    _conn: Tlg.Connection

    connection = GObject.Property(type=Tlg.Connection, flags=GObject.ParamFlags.READWRITE|GObject.ParamFlags.CONSTRUCT)

    def __init__(self, conn: Tlg.Connection):
        super().__init__(connection=conn)

    def do_clicked(self, *args):
        self.emit('selected', self.props.connection)
    
